### PR TITLE
Fix custom object loading when opening a project

### PR DESCRIPTION
**Pull request by D8H**

### CODE REVIEW

File: Core/GDCore/Project/EventsFunctionsExtension.cpp [hunk 1]
```diff
@@ -185,11 +189,93 @@ void EventsFunctionsExtension::UnserializeExtensionImplementationFrom(
...
+
+      if (!isDependentFromRemainingEventsBasedObjects(
+              eventsBasedObjectElement)) {
+        loadOrderEventsBasedObjectNames.push_back(eventsBasedObjectName);
+        remainingEventsBasedObjectNames.erase(
+            remainingEventsBasedObjectNames.begin() + i);
+        i--;
+        foundAnyEventsBasedObject = true;
+      }
+    }
```
Comment (4ian) [inline Core/GDCore/Project/EventsFunctionsExtension.cpp:276]: I think an object A depending on B and B depending on A will result in an infinite loop?

File: Core/GDCore/Project/Project.cpp [hunk 2]
```diff
@@ -830,7 +830,7 @@ void Project::UnserializeFrom(const SerializerElement& element) {
   eventsFunctionsExtensionsElement.ConsiderAsArrayOf(
       "eventsFunctionsExtension");
   // First, only unserialize behaviors and objects names.
-  // As event based objects can contains CustomObject and Custom Object,
+  // As event based objects can contains custom behaviors and custom objects,
```
Comment (4ian) [inline Core/GDCore/Project/Project.cpp:833]: 👍

File: Core/tests/ObjectSerialization.cpp [hunk 3]
```diff
@@ -262,4 +262,137 @@ TEST_CASE("ObjectSerialization", "[common]") {
...
+    CheckSpriteConfiguration(spriteConfiguration);
+  }
+
+  SECTION("Save and load a project with custom object dependencies inside an extension") {
+    gd::Platform platform;
+    gd::Project writtenProject;
+    SetupProjectWithDummyPlatform(writtenProject, platform);
+
+
+  std::cout << "TEST START" << std::endl;
```
Comment (4ian) [inline Core/tests/ObjectSerialization.cpp:339]: To remove

File: Core/GDCore/Project/EventsFunctionsExtension.cpp (hunk 1 above)
Comment (D8H) [reply Core/GDCore/Project/EventsFunctionsExtension.cpp:276]: The editor doesn't allow it, but they would only be ignored and not loaded which is not the best but still fine I guess.
We could log what remains in `remainingEventsBasedObjectNames` at the end of the loop.

**Pull request merged**